### DISCUSSION TITLE
Big JSON Payloads - is this technique ok?

### DISCUSSION THREAD

**Discussion by Xenossolitarius** [nuxt/nuxt#28764] [Questions]
In my project I stumbled on "needing" to have one giant JSON payload. Around 1.2 MB. I wanted to have it bolted in HTML, SSR'd. Let's just go with the flow and not debate if I should even send it to the frontend. Bad practice. But issue arises when it's smaller also of how complex it would be. The issue is with nuxt/devalue where its trying to transform the payload, ofc to stop us from getting attacked by the XSS, And it takes time. It takes 32ms for example which is almost as long as rendering time. This json payload is not static. Changes every couple of minutes.
I wanted to trick the nuxt/devalue with this code:
```
export default definePayloadPlugin((nuxt) => {
  definePayloadReducer('JSONPayload', (value) => {
    if (!(value instanceof JSONPayload)) return false
    return value.toString()
  })
  definePayloadReviver('JSONPayload', (value) => {
    return new JSONPayload().fromString(value)
  })
})
```
basically tricking it into a string, circumventing devalue slow performance. But then it took 66 ms. Twice as long. What's up? So it seems the devalue is extra careful around the strings, so the parsing lasts.

Eventually I settled on this code:

```
export default definePayloadPlugin((nuxt) => {
  definePayloadReducer('JSONPayload', (value) => {
    if (!(value instanceof JSONPayload)) return false

    nuxt.runWithContext(() => {
      useHead({
        script: [
          {
            tagPosition: 'bodyClose',
            innerHTML: 'window.__PLD__ =' + JSON.stringify(value.payload),
          },
        ],
      })
    })

    return 'pld'
  })
  definePayloadReviver('JSONPayload', (value) => {
    const pld= new JSONPayload()
    pld.payload = window.__PLD__
    return pld
  })
})
```
This brought down the payloadRendering time to a  staggering 7ms. Still quite long but this was managable.

So in general, couple of questions here:
Is there a cleaner way around this issue?
Does this feature exists or is it a good candidate for new feature?
What will be this feature? A boolean that skips the "stringify"? Virtual modules linking to external JSON's ? 
Does this incur any memory leaks (i checked it doesn't)?
This is definitely not safe but how bad is it ?

**Comment by viii47** [comment]
> Is there a cleaner way around this issue? [...] Does this feature exists or is it a good candidate for new feature?

Depends on the use-case. It'd be easier to roughly know what the object needs to do to. Parsing a large object in itself is always staggeringly slow if it hits a certain size or over (>1MB as mentioned). Is the raw size 1.2 Mb or more?
